axelor-account/src/main/java/com/axelor/apps/account/service/AnalyticDistributionTemplateServiceImpl.java

```diff
@@ -86,4 +124,95 @@ public AnalyticDistributionTemplate createDistributionTemplateFromAccount(Accoun
     }
     return analyticDistributionTemplate;
   }
+
+  @Override
+  public void checkAnalyticAccounts(AnalyticDistributionTemplate analyticDistributionTemplate)
+      throws AxelorException {
+    if (analyticDistributionTemplate != null
+        && CollectionUtils.isNotEmpty(
+            analyticDistributionTemplate.getAnalyticDistributionLineList())) {
+      for (AnalyticDistributionLine analyticDistributionLine :
+          analyticDistributionTemplate.getAnalyticDistributionLineList()) {
+        if (analyticDistributionLine.getAnalyticAccount() == null) {
+          throw new AxelorException(
+              TraceBackRepository.CATEGORY_CONFIGURATION_ERROR,
+              I18n.get(IExceptionMessage.FIXED_ASSET_ANALYTIC_ACCOUNT_MISSING));
+        }
+      }
+    }
+  }
+
+  protected AnalyticDistributionTemplate personalizeAnalyticTemplateFromConfig(
+      AnalyticDistributionTemplate analyticDistributionTemplate,
+      AnalyticDistributionTemplate newAnalyticDistributionTemplate,
+      Company company)
+      throws AxelorException {
+    AccountConfig accountConfig = accountConfigService.getAccountConfig(company);
+    newAnalyticDistributionTemplate.setCompany(company);
+    newAnalyticDistributionTemplate.setName("Template - ");
+
+    newAnalyticDistributionTemplate.setIsSpecific(true);
+
+    for (AnalyticAxisByCompany analyticAxisByCompany :
+        accountConfig.getAnalyticAxisByCompanyList()) {
+      if (!fillNewAnalyticDistributionTemplate(
+          analyticAxisByCompany,
+          analyticDistributionTemplate,
+          newAnalyticDistributionTemplate,
+          accountConfig)) {
+        newAnalyticDistributionTemplate =
+            personalizeAnalyticDistributionLine(
```

Comment:
it would be clearer to refactor this:
- fillNewAnalyticDistributionTemplate should be replaced by a method returning found analyticDistributionLine
- if analyticDistributionLine is not found we can use default values to call the method `personalizeAnalyticDistributionLine` (you can use Optional for this).